File: api/rmbg.js
import { Client } from "@gradio/client";

// Helper for artificial delay
const delay = (ms) => new Promise(res => setTimeout(res, ms));

export const config = {
  api: { bodyParser: false }
};

export default async function handler(req, res) {
  // Fix for the 405 error: Log what method is actually hitting the server
  if (req.method !== "POST") {
    console.log("Received method:", req.method);
    return res.status(405).json({ error: `Expected POST, got ${req.method}` });
  }

  try {
    // 1. Collect input (Input "Throttling" by waiting)
    const chunks = [];
    for await (const chunk of req) { chunks.push(chunk); }
    const buffer = Buffer.concat(chunks);
    
    await delay(2000); // Wait 2 seconds

    // 2. AI Processing
    const client = await Client.connect("briaai/BRIA-RMBG-2.0");
    const result = await client.predict("/image", { 
      image: new Blob([buffer], { type: "image/png" }) 
    });

    // 3. Prepare Output
    const file = result.data[1];
    const imgResponse = await fetch(file.url);
    const finalBuffer = Buffer.from(await imgResponse.arrayBuffer());

    await delay(5000); // Wait another 2 seconds

    // 4. Send response
    res.setHeader("Content-Type", "image/png");
    res.send(finalBuffer);

  } catch (e) {
    console.error(e);
    res.status(500).json({ error: e.message });
  }
      }

File: api/rmbg.py
# api/rmbg.py
from io import BytesIO
from gradio_client import Client

GRADIO_SPACE = "https://briaai-bria-rmbg-1-4.hf.space/--replicas/sc92z/"

def handler(request):
    """
    Vercel serverless handler
    """
    try:
        img_bytes = request.data
        if not img_bytes:
            return {"error": "No data received"}, 400

        client = Client(GRADIO_SPACE)
        result_path = client.predict(BytesIO(img_bytes), api_name="/predict")

        if not isinstance(result_path, str):
            return {"error": "Unexpected result type", "result": str(result_path)}, 500

        # Download output
        output_buffer = BytesIO()
        client.download(result_path, output_buffer)
        output_buffer.seek(0)

        # Return raw bytes
        return output_buffer.getvalue(), 200, {"Content-Type": "image/png"}

    except Exception as e:
        return {"error": str(e)}, 500
